### DISCUSSION TITLE
Sonoff Zigbee Bridge Pro restart and coordinator restart

### DISCUSSION THREAD

**Discussion by Saur0o0n** [arendst/Tasmota#17213] [Q&A]
Hi,

Few times now, while doing zigbee2mqtt restart I had a problem with herdsman connecting to coordinator on Zigbee bridge. The TCP connection was ok, but there was no response from coordinator.
But my question is not on the issue itself, but does the coordinator is also restarted while doing Tasmota restart of Zonoff zigbee bridge pro? By restart I mean pushing restart button on web interface. The question is because, after this type of restart the connection issue still persist - but if I do hard reset (power unplug/plug) - it gets back to normal, working condition.

Log from non working z2mqtt - just as an addition...
Zigbee2MQTT:info  2022-11-29 10:37:35: Logging to console and directory: '/home/hassio/zigbee2mqtt/data/log/2022-11-29.10-37-35' filename: log.txt
Zigbee2MQTT:info  2022-11-29 10:37:35: Starting Zigbee2MQTT version 1.28.2 (commit #e66fe853)
Zigbee2MQTT:info  2022-11-29 10:37:35: Starting zigbee-herdsman (0.14.68)
Zigbee2MQTT:error 2022-11-29 10:37:54: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2022-11-29 10:37:54: Failed to start zigbee
Zigbee2MQTT:error 2022-11-29 10:37:54: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2022-11-29 10:37:54: Exiting...
Zigbee2MQTT:error 2022-11-29 10:37:54: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.start (/home/hassio/zigbee2mqtt/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:103:27)
    at Controller.start (/home/hassio/zigbee2mqtt/node_modules/zigbee-herdsman/src/controller/controller.ts:132:29)
    at Zigbee.start (/home/hassio/zigbee2mqtt/lib/zigbee.ts:58:27)
    at Controller.start (/home/hassio/zigbee2mqtt/lib/controller.ts:101:27)
    at start (/home/hassio/zigbee2mqtt/index.js:109:5)

**Comment by s-hadinger** [comment]
I confirm that restarting Tasmota doesn't restart the Zigbee MCU. Zigbee2MQTT needs to send the MCU restart sequence. Tasmota has no knowledge on how to restart the MCU (except in Zigbee2Tasmota).
Any other question should be adressed to Zigbee2MQTT.

**Reply by Saur0o0n** [reply]
Thanks for confirming what I've supposed after observation. Z2MQTT cannot connect to Zigbee MCU - so it also cannot send any command. There is no way to restart that MCU on electric level?

**Reply by s-hadinger** [reply]
Yes there is a way to force reset on the MCU, driving GPIO15 low. You can assigne a Relay on GPIO15 and turn it OFF then ON again to force a hardware reset.
Zigbee2MQTT could also send a software reset by sending a sequence of 256x 0xFF.

**Reply by baszied** [reply]
Hi, 

Kind of a newbee in this area, flashed to the Sonoff ZB Bridge Pro with tasmota32-zbbrdgpro.bin and think I'm facing some kind of the same issue...  With autoconfiguration option 'Sonoff ZBPro', I'm able to see the 'Zigbee Permit Join' and 'Map' buttons in the main menu and I can switch zigbee devices (ZbSend). In this config however I dont see a TCP listener started.  Switching to autoconfig 'Sonoff ZBPro TCP' does give me a listener but doesn't allow me to let Zigbee devices added/controlled...

As said, I'm a newbee... Could you provide some more detailed instructions what is exactly done when you say:
".. assign a Relay on GPIO15 and turn it OFF then ON again to force a hardware reset." ?

Many thanks!

**Reply by sfromis** [reply]
You can either run in Z2T mode where Tasmota acts a coordinator, managing Zigbee devices, controllable via MQTT messages. Or you can run in TCP mode having ZHA or Z2M in the coordinator role, with Tasmota only providing the connection. What you see looks like it is working as intended.

The thing about gpio15 is that you can use a relay config to have that pin high or low, and then you can force a reset of the Zigbee chip using a pulse. This is based on the specific hardware connecting the reset pin of the Zigbee chip to the MCU. You'll likely not need this, or only rarely.

**Reply by baszied** [reply]
Thanks, very helpfull info; I'll stay with my current setup that works fine for me;
